src/main/java/com/google/devtools/build/lib/rules/objc/ObjcCommon.java

```diff
@@ -661,6 +662,22 @@ static boolean shouldUseObjcModules(RuleContext ruleContext) {
     return false;
   }
 
+  static String getClangModuleName(RuleContext ruleContext) {
+    if (ruleContext.attributes().has("module_name", Type.STRING)) {
+      return ruleContext.attributes().get("module_name", Type.STRING);
```

Comment:
I had issues moving this to a Support class since this method is used by both BundleSupport and IntermediateArtifacts.

With regards to validation. I think the modulemap langauge supports basically any arbitrary string as a module name, however one can't use "@import" syntax if uses a special objective-c character, so not sure if we have to validate. I'll add a comment to that extent.